src/utils/train.py

```diff
@@ -10,13 +10,35 @@
 from torch.utils.data import Dataset, DataLoader
 from tqdm.auto import tqdm
 from transformers import get_linear_schedule_with_warmup
+from torch.nn import CTCLoss
+import numpy as np
+from torch.nn.utils.rnn import pad_sequence
 
 from src.utils.common import set_deterministic_mode, dict_to_device
 from src.utils.config import TrainingConfig
+from src.utils.common import char2int
+from src.datasets.vtr_dataset import VTRDatasetOCR
+
 
 WANDB_PROJECT_NAME = "visual-text"
 
 
+def compute_ctc_loss(criterion: torch.nn.modules.loss.CTCLoss, logits: torch.Tensor, texts: list, char_set: set):
```

Comment:
After the order of the dimensions above changes, they will need to be changed in this function as well. Keep in mind that the loss has no option `batch_first`. https://pytorch.org/docs/stable/generated/torch.nn.CTCLoss.html

The loss function takes a tensor of size (T, N, C) or (T, C) where T = sequence length, N = batch size, C = num classes.

Double check that the prediction tensors are correctly related to the ground truth texts